tools/build/docker/Dockerfile

```diff
@@ -29,27 +38,46 @@ RUN \
     adduser -D -u ${LRR_UID} -G koyomi koyomi; \
 fi
 
-WORKDIR /home/koyomi/lanraragi
 
-#Copy wsl.conf to /etc for extra WSL compatibility 
-COPY /tools/build/windows/wsl.conf /etc/wsl.conf
+# we use s6-overlay-nobin to just pull in the s6-overlay arch agnostic (shell)
+# components, since we apk install the binaries of s6 later which are arch specific
+ADD https://github.com/just-containers/s6-overlay/releases/download/${S6_OVERLAY_RELEASE}/s6-overlay-nobin.tar.gz /tmp/s6-overlay-nobin.tar.gz
+RUN tar -C / -xzf /tmp/s6-overlay-nobin.tar.gz && rm -f /tmp/s6-overlay-nobin.tar.gz
+
+
+WORKDIR /home/koyomi/lanraragi
 
 #Copy cpanfile and install script before copying the entire context
 #This allows for Docker cache to preserve cpan dependencies
-COPY --chown=koyomi:koyomi /tools/cpanfile /tools/install.pl /tools/build/docker/* tools/
+COPY --chown=koyomi:koyomi /tools/cpanfile /tools/install.pl tools/
 COPY --chown=koyomi:koyomi /package.json package.json
 
-# Make scripts executable + Run the install script as root
-RUN chmod +x ./tools/install-everything.sh ./tools/entrypoint.sh && sh ./tools/install-everything.sh
+# Install deps (we install Linux::Inotify2 manually since it's not in the base cpanfile as it doesn't build on macOS)
+RUN apk add --update perl perl-io-socket-ssl perl-dev redis libarchive-dev libbz2 openssl-dev zlib-dev \
+imagemagick imagemagick-perlmagick libwebp-tools \
+g++ make pkgconf gnupg wget curl nodejs nodejs-npm \
+shadow s6 s6-portable-utils && \
+curl -L https://cpanmin.us | perl - App::cpanminus && \
+cd tools && cpanm --notest --installdeps . -M https://cpan.metacpan.org && cd .. && \
+npm run lanraragi-installer install-front && \
+apk del perl-dev g++ make gnupg wget curl nodejs nodejs-npm openssl-dev && \
+rm -rf /root/.cpanm/* /root/.npm/ /usr/local/share/man/* node_modules /var/cache/apk/*
 
-#Copy remaining LRR files from context  
+#Copy remaining LRR files from context
+# consider chowning in s6 setup scripts instead
 COPY --chown=koyomi:koyomi /lib lib
 COPY --chown=koyomi:koyomi /public public
 COPY --chown=koyomi:koyomi /script script
 COPY --chown=koyomi:koyomi /templates templates
 COPY --chown=koyomi:koyomi /tests tests
 COPY --chown=koyomi:koyomi /lrr.conf lrr.conf
+COPY /tools/build/docker/redis.conf /etc/redis.conf
```

Comment:
Forgot about this, but the WSL version is currently set to use the redis.conf in `/home/koyomi/lanraragi/tools/build/docker/redis.conf` specifically. See https://github.com/Difegue/Karen/blob/1aa5e881b25473391737f64bc50ad64957ac6898/Karen/Interop/WslDistro.cs#L168 .  

Moving the file to `/etc/redis.conf` is a saner default I think, but I'll have to update the windows code.  
(Gotta update the comments as well, we don't use any form of init in the WSL version)

Reply:
I don't think any of the other changes here break the WSL version, so that's great! 🤠

Reply:
I'd forgotten that redis-server doesn't load defaults automatically from `/etc`, so it might be better to move the conf file back to `build/docker` so we don't forget about that again.. At least adding a comment would be worthwhile I think.